lib/streamlit/elements/widgets/file_uploader.py

```diff
@@ -442,21 +446,20 @@ def _file_uploader(
             label_visibility
         )
 
-        file_uploader_state = serde.serialize(widget_state.value)
-        uploaded_file_info = file_uploader_state.uploaded_file_info
-        if ctx is not None and len(uploaded_file_info) != 0:
-            newest_file_id = file_uploader_state.max_file_id
-            active_file_ids = [f.id for f in uploaded_file_info]
-
-            ctx.uploaded_file_mgr.remove_orphaned_files(
-                session_id=ctx.session_id,
-                widget_id=file_uploader_proto.id,
-                newest_file_id=newest_file_id,
-                active_file_ids=active_file_ids,
-            )
-
         self.dg._enqueue("file_uploader", file_uploader_proto)
-        return widget_state.value
+
+        filtered_value: Union[UploadedFile, List[UploadedFile], None]
+
+        if isinstance(widget_state.value, DeletedFile):
+            filtered_value = None
+        elif isinstance(widget_state.value, list):
+            filtered_value = [
+                f for f in widget_state.value if not isinstance(f, DeletedFile)
+            ]
+        else:
+            filtered_value = widget_state.value
+
+        return filtered_value
```

Comment:
nit: I think it'd be fine to return the value directly from each branch of the if/elif/else statement rather than save it to an intermediate variable that's returned immediately

Reply:
done